portfolio/src/main/webapp/script.js

```diff
@@ -1,3 +1,72 @@
+
+/** Shows a popup window. */
+function showPopup(button) {
+  const modal = document.getElementById('my-popup');
+  const content = document.getElementById('popup-content');
+
+  removeLastChild(content);
+
+  switch (button.id) {
+    case '1':
+      document.getElementById('picture').src = 'images/pickleball.jpg';
+      appendElement('h3', 'Sports and Fitness', content);
+      appendElement(
+          'p',
+          'I love doing yoga, hiking, and have recently started playing pickleball. Fun fact: I fenced competitively for 7 years',
+          content);
+      break;
+    case '2':
+      document.getElementById('picture').src = 'images/nelson.jpg';
+      appendElement('h3', 'Art and Drawing', content);
+      appendElement(
+          'p',
+          'I enjoy visiting art museums and drawing in my free time. Right now, I like drawing realistic portrait style pencil portraits. One of my favorite museums is The Nelson-Atkins Museum of Art here in Kansas City.',
+          content);
+      break;
+
+    case '3':
+      document.getElementById('picture').src = 'images/summersalt.jpg';
+      appendElement('h3', 'Music and Concerts', content);
+      appendElement(
+          'p',
+          'This image is from the last concert I went to at a small venue called EXIT/IN in Nashville. I love finding new songs and artists to add to my Spotify playlists.',
+          content);
```

Comment:
Lots of repetition of these lines here. I think it might look a little nicer if you used more local variables in this method that you assigned values to in each case statement, e.g.

let imagePath;
let headerText;
let captionText;

Then in each case statement..

imagePath =
headerText =
captionText =

Then after the switch:
```
document.getElementById('picture').src = imagePath;
appendElement('h3', headerText, content);
appendElement('p', captionText, content);
```

Reply:
This definitely looks a lot cleaner and more readable!